tests/unit/spawn_test.cc

```diff
@@ -116,3 +116,23 @@ SEASTAR_TEST_CASE(test_spawn_input) {
         });
     });
 }
+
+SEASTAR_TEST_CASE(test_spawn_kill) {
+    const char* sleep_cmd = "/bin/sleep";
+    // sleep for 10s, but terminate it right away.
+    return spawn_process(sleep_cmd, {.argv = {sleep_cmd, "10"}}).then([] (auto process) {
+        auto start = std::chrono::high_resolution_clock::now();
+        return do_with(std::move(process), [](auto& p) {
+            p.terminate();
+            return p.wait();
+        }).then([start](experimental::process::wait_status wait_status) {
+            auto* wait_signaled = std::get_if<experimental::process::wait_signaled>(&wait_status);
+            BOOST_REQUIRE(wait_signaled != nullptr);
+            BOOST_CHECK_EQUAL(wait_signaled->terminating_signal, SIGTERM);
+            // sleep should be terminated in 10ms
```

Comment:
How is the number calculated or it's just an estimation?

Reply:
it's a rough estimation. 10ms is more than enough for the RTT of local system without involving networking or disk i/o.

Reply:
It takes nearly 20ms on my machine(ArchLinux, running in docker), so weird.

> /path/to/spawn_test.cc(135): error: in "test_spawn_kill": check ms <= 10 has failed [20 > 10]

Reply:
20ms is indeed unexpected. what's the latency between a `kill(2)` call and the completion of `waitpid()` ? i created a tiny tool, see https://godbolt.org/z/rdraqsaMG

and my output is like
```console
$  g++ test_fork.cc
$ ./a.out sleep 10
child killed by signal 15
duration: 000:000:275165
sleep returned -1
```

Reply:
looks quite normal:

```txt
% clang++ kill.cc
% ./a.out sleep 10
child killed by signal 15
duration: 000:000:102323
sleep returned -1
```

Reply:
please feel free to file an issue. or if you have a theory explaining this, a PR would be better!

i don't have an explanation for the discrepancy between 102323ns and 20ms. reactor's period of polling should not incur such a huge latency. IMHO, both linux and `sleep` should be very responsive under normal load.

Reply:
sure. I'll make a deep investigation this weekend.